src/hooks/useCioClient.ts

```diff
@@ -4,14 +4,18 @@ import { useMemo } from 'react';
 import { getCioClient } from '../utils';
 
 export type CioClientConfig = { apiKey?: string; cioJsClient?: ConstructorIOClient };
-type UseCioClient = (cioClientConfig: CioClientConfig) => ConstructorIOClient | undefined;
+type UseCioClient = (cioClientConfig: CioClientConfig) => ConstructorIOClient | never;
```

Comment:
how is dis:
`ConstructorIOClient | never`

different than this:
`ConstructorIOClient`

Reply:
ConstructorIOClient | never means that the function might never get to it's last line and even return undefined. Basically I'm communicating the `throw new Error() `line through typescript by specifying type never meaning it might throw and error and never return anything even undefined @deomsj